### PR TITLE
Fix Frecpe_S/V and Frsqrte_S/V (full FP emu.). Add Sse Opt. & SoftFloat Impl. for Fcmeq/ge/gt/le/lt_S/V (Reg & Zero), Faddp_S/V, Fmaxp_V, Fminp_V Inst.; add Sse Opt. for Shll_V, S/Ushll_V Inst.; improve Sse Opt. for Xtn_V Inst.. Add Tests.

### PR DESCRIPTION
**Pull request by LDj3SNuD**
### **Added Fast (Intrinsics) & Slow (SoftFloat) paths:**

_InstEmitSimdCmp:_

**Fcmeq, Fcmge, Fcmgt, Fcmle, Fcmlt (Scalar & Vector, Reg & Zero):**
- Max Sse Req.: Sse2
- Types Covered: All

_InstEmitSimdArithmetic:_

**Faddp_S:**
- Max Sse Req.: Sse3
- Types Covered: All

**Faddp_V;**
**Fmaxp_V, Fminp_V:**
- Max Sse Req.: Sse2
- Types Covered: All

### **Added Fast (Intrinsics) paths:**

_InstEmitSimdShift:_

**Shll_V;**
**Sshll_V, Ushll_V:**
- Max Sse Req.: Sse41
- Types Covered: All

_InstEmitSimdMove:_

**Xtn_V (improved):**
- Max Sse Req.: Ssse3
- Types Covered: All

**Tests provided for all instructions involved.**

Nits.

### CODE REVIEW

**Comment by LDj3SNuD** [comment]
Fix Frecpe_S/V and Frsqrte_S/V (full FP emu.). Add 4 FP Tests.

Tests provided for all instructions involved.

Nits.

**Comment by LDj3SNuD** [comment]
Fix Frecpe_S/V and Frsqrte_S/V (full FP emu.). Add 4 FP Tests.

Tests provided for all instructions involved.

Nits.

**Comment by LDj3SNuD** [comment]
Fast (Intrinsics) & Slow (SoftFloat) paths:
Fabd_S, Fabd_V:
Max Sse Req.: Sse2
Types Covered: All

Tests provided for all instructions involved.

Nits.

**Comment by LDj3SNuD** [comment]
Fast (Intrinsics):
Frecpe_S, Frecpe_V;
Frsqrte_S, Frsqrte_V:
Max Sse Req.: Sse
Types Covered: S

Re-tested all instructions involved.

File: ChocolArm64/Instructions/SoftFloat.cs [hunk 1]
```diff
@@ -2249,6 +2629,11 @@ private static double FPNeg(this double value)
             return -value;
         }
 
+        private static double ZerosOrOnes(bool zeros)
+        {
+            return BitConverter.Int64BitsToDouble(!zeros ? 0L : -1L);
```
Comment (gdkchan) [inline ChocolArm64/Instructions/SoftFloat.cs]: It's a bit weird to me that a function called `ZerosOrOnes` with a `zeros` arguments, returns zero when the argument is *false* and not true.

Comment (LDj3SNuD) [reply ChocolArm64/Instructions/SoftFloat.cs]: Indeed; I will take this into account in a later PR.
Thanks.

**Pull request merged**